### PR TITLE
Make QuarkusMock work with REST Client Reactive interfaces

### PR DESCRIPTION
**Pull request by michalszynkiewicz**

### CODE REVIEW

File: docs/src/main/asciidoc/rest-client-reactive.adoc [hunk 1]
```diff
@@ -858,11 +858,119 @@ quarkus.rest-client.logging.body-limit=50
 quarkus.log.category."org.jboss.resteasy.reactive.client.logging".level=DEBUG
 ----
 
+== Mocking the client for tests
+If you use a client injected with the `@RestClient` annotation, you can easily mock it for tests.
+You can do it with Mockito's `@InjectMock` or with `QuarkusMock`.
+
+This section shows how to replace your client with a mock. If you would like to get a more in-depth understanding of how mocking works in Quarkus, see the blog post on https://quarkus.io/blog/mocking/[Mocking CDI beans].
+
+NOTE: Mocking does not work in native tests.
```
Comment (geoand) [inline docs/src/main/asciidoc/rest-client-reactive.adoc:867]:
```suggestion
NOTE: Mocking does not work when using `@NativeImageTest` or `@QuarkusIntegrationTest`.
```

**Review by geoand — APPROVED** [review]
Nice!

**Comment by geoand** [comment]
I assume we do want this in `2.7`, right?

File: docs/src/main/asciidoc/rest-client-reactive.adoc (hunk 1 above)
Comment (michalszynkiewicz) [reply docs/src/main/asciidoc/rest-client-reactive.adoc]: done

File: docs/src/main/asciidoc/rest-client-reactive.adoc [hunk 2]
```diff
@@ -858,11 +858,119 @@ quarkus.rest-client.logging.body-limit=50
...
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+
+@QuarkusTest
+public class InjectMockTest {
+
+    @InjectMock
+    ClientWithExceptionMapper mock;
```
Comment (mkouba) [inline docs/src/main/asciidoc/rest-client-reactive.adoc:919]: Where does the `ClientWithExceptionMapper` come from?

File: core/runtime/src/main/java/io/quarkus/runtime/MockedThroughWrapper.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package io.quarkus.runtime;
+
+/**
+ * Usually, QuarkusMock mocking replaces a concrete CDI bean with a mock.
```
Comment (mkouba) [inline core/runtime/src/main/java/io/quarkus/runtime/MockedThroughWrapper.java:4]: In fact, we don't replace a concrete bean instance but a "delegating instance" of a client proxy. That's why only normal scopes are supported.

File: docs/src/main/asciidoc/rest-client-reactive.adoc (hunk 2 above)
Comment (michalszynkiewicz) [reply docs/src/main/asciidoc/rest-client-reactive.adoc:919]: oops, fixing

File: core/runtime/src/main/java/io/quarkus/runtime/MockedThroughWrapper.java (hunk 3 above)
Comment (michalszynkiewicz) [reply core/runtime/src/main/java/io/quarkus/runtime/MockedThroughWrapper.java]: I tried to add it to the javadoc :)

**Review by mkouba — APPROVED** [review]

**Pull request merged**